master/buildbot/data/changes.py

```diff
@@ -78,8 +78,13 @@ def get(self, resultSpec, kwargs):
             else:
                 changes = []
         else:
-            changes = yield self.master.db.changes.getChanges()
-
+            # this special case is useful and implemented by the dbapi
+            # so give it a boost
+            if all((resultSpec.limit, resultSpec.offset is None,
+                    resultSpec.order == ['-changeid'])):
```

Comment:
looks like the pep8 bug is actually fixed now. switch to the more logical expression